engine/apps/schedules/tests/test_notify_about_empty_shifts_in_schedule.py

```diff
@@ -0,0 +1,170 @@
+        channel="channel",
+        prev_ical_file_overrides=None,
+        cached_ical_file_overrides=None,
+    )
+
+    now = timezone.now().replace(microsecond=0)
+    start_date = now - datetime.timedelta(days=7, minutes=1)
+    data = {
+        "start": start_date,
+        "rotation_start": start_date,
+        "duration": datetime.timedelta(seconds=3600 * 24),
+        "priority_level": 1,
+        "frequency": CustomOnCallShift.FREQUENCY_DAILY,
+        "schedule": schedule,
+    }
+    on_call_shift = make_on_call_shift(
+        organization=organization, shift_type=CustomOnCallShift.TYPE_ROLLING_USERS_EVENT, **data
+    )
+    on_call_shift.add_rolling_users([[user1]])
+    schedule.refresh_ical_file()
+
+    empty_shifts_report_sent_at = schedule.empty_shifts_report_sent_at
+
+    with patch("apps.slack.slack_client.SlackClientWithErrorHandling.api_call") as mock_slack_api_call:
+        notify_about_empty_shifts_in_schedule(schedule.pk)
+
+    assert not mock_slack_api_call.called
+
+    schedule.refresh_from_db()
+    assert empty_shifts_report_sent_at != schedule.empty_shifts_report_sent_at
+
+
+@pytest.mark.django_db
+def test_empty_shifts_trigger_notification(
+    make_organization_and_user_with_slack_identities,
+    make_user,
+    make_schedule,
+    make_on_call_shift,
+):
+    organization, _, _, _ = make_organization_and_user_with_slack_identities()
+    user1 = make_user(organization=organization, username="user1", role=LegacyAccessControlRole.VIEWER)
+    # clear users pks <-> organization cache (persisting between tests)
+    memoized_users_in_ical.cache_clear()
+
+    schedule = make_schedule(
+        organization,
+        schedule_class=OnCallScheduleWeb,
+        name="test_schedule",
+        channel="channel",
+        prev_ical_file_overrides=None,
+        cached_ical_file_overrides=None,
+    )
+
+    now = timezone.now().replace(microsecond=0)
+    start_date = now - datetime.timedelta(days=7, minutes=1)
+    data = {
+        "start": start_date,
+        "rotation_start": start_date,
+        "duration": datetime.timedelta(seconds=3600 * 24),
+        "priority_level": 1,
+        "frequency": CustomOnCallShift.FREQUENCY_DAILY,
+        "schedule": schedule,
+    }
+    on_call_shift = make_on_call_shift(
+        organization=organization, shift_type=CustomOnCallShift.TYPE_ROLLING_USERS_EVENT, **data
+    )
+    on_call_shift.add_rolling_users([[user1]])
+    schedule.refresh_ical_file()
+
+    empty_shifts_report_sent_at = schedule.empty_shifts_report_sent_at
+
+    with patch("apps.slack.slack_client.SlackClientWithErrorHandling.api_call") as mock_slack_api_call:
+        notify_about_empty_shifts_in_schedule(schedule.pk)
+
+    assert mock_slack_api_call.called
+
+    schedule.refresh_from_db()
+    assert empty_shifts_report_sent_at != schedule.empty_shifts_report_sent_at
+    assert schedule.has_empty_shifts
+
+
+@pytest.mark.django_db
+def test_empty_non_empty_shifts_trigger_notification(
```

Comment:
I guess this is the original behavior, but wondering if this should be a warning in the schedule UI but not triggering a notification (since there is still an active user for the shift). In any case, something to discuss later.

Reply:
Agree, we should discuss it later. I think, the main point is make customer know that something is wrong with one of their shifts despite the presence of an active user.